### PR TITLE
Ontology extender + cdr indexer

### PR DESCRIPTION
**Pull request by maxaalexeeva**
OntologyExtender app:
- reads in ontology node files (one file per node)
- gets hyponyms and co-hyponyms for every example
- writes files containing both existing and newly added (co-)hyponym examples to a new directory
- this is a basic implementation, which can be improved (after initial evaluation?) by a) filtering examples using vector similarity b) searching for co-hyponyms for the newly-added hyponym examples (this step can be easily added, but it may make sense to add that after some filtering is implemented)  

IndexCDRs app:
- reads CDR files
- writes the corresponding Document and text files
- creates an index for the cdr dataset

Other changes:
- updated taxero deps to more recent versions 
- small updates to various methods based on the newer versions of dependencies
- lemmatization made optional while looking for hypo-/hyper-/etc-nyms

### CODE REVIEW

File: src/main/scala/org/clulab/taxero/TaxonomyReader.scala [hunk 1]
```diff
@@ -49,38 +49,38 @@ class TaxonomyReader(
...
-    rankMatches(lemmas, matches) 
+    rankMatches(items, matches)
   }
 
   def getExpandedHypernyms(pattern: Seq[String], n: Int): Seq[ScoredMatch] = {
     // start query set with the provided query
     val allQueries = mutable.HashSet(pattern)
     // add the n closest cohyponyms to the query set
-    allQueries ++= getRankedCohyponyms(pattern).take(n).map(_.result)
+    allQueries ++= getRankedCohyponyms(pattern, lemmatize = true).take(n).map(_.result)
```
Comment (BeckySharp) [inline src/main/scala/org/clulab/taxero/TaxonomyReader.scala]: in all these places where you have `lemmatize=true`, we should prob set one variable above, that is loaded from a config and use it in these places, to not adjust existing functionality of the webapp

File: src/main/scala/org/clulab/taxero/TaxonomyReader.scala [hunk 2]
```diff
@@ -169,38 +169,46 @@ class TaxonomyReader(
...
-      mkExtractorsFromRules(tokens, rules.mkString)
+      mkExtractorsFromRules(tokens, rules.mkString, lemmatize)
     }
   }
 
-  def mkExtractorsFromRules(tokens: Seq[String], rules: String): Seq[Extractor] = {
-    val variables = Map("query" -> mkLemmaPattern(tokens))
-    extractorEngine.ruleReader.compileRuleFile(rules.mkString, variables)
+  def mkExtractorsFromRules(tokens: Seq[String], rules: String, lemmatize: Boolean): Seq[Extractor] = {
+    val variables = if (lemmatize == true) {
```
Comment (BeckySharp) [inline src/main/scala/org/clulab/taxero/TaxonomyReader.scala]: here you can just say `if (lemmatize)`

File: src/main/scala/org/clulab/taxero/TaxonomyReader.scala (hunk 1 above)
Comment (BeckySharp) [reply src/main/scala/org/clulab/taxero/TaxonomyReader.scala]: hmmm -- though I'm ok with the just saying `=true` in the HomeController, so maybe that's fine, but why are you enforcing it here -- this is the only place, right?

**Review by BeckySharp — COMMENTED** [review]
I have a question, but otherwise looks great!

**Comment by maxaalexeeva** [comment]
@BeckySharp, this should be ready to review.

**Review by BeckySharp — APPROVED** [review]
thanks!

**Pull request merged**